src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -145,6 +156,14 @@ trait IterableOps[+A] extends Any {
   *  as the one they are invoked on.
   */
 trait IterableMonoTransforms[+A, +Repr] extends Any {
+  def filter(p: A => Boolean): Repr
```

Comment:
The way we solved that is to make `Range` extend `IterableMonoTransforms[Int, Seq[Int]]` and then covariantly override the return type of the other methods of `IterableMonoTransforms` (e.g. `take`).